src/ray/raylet/node_manager.cc

```diff
@@ -748,21 +768,28 @@ void NodeManager::HeartbeatAdded(const ClientID &client_id,
 
   SchedulingResources &remote_resources = it->second;
 
-  ResourceSet remote_total(VectorFromProtobuf(heartbeat_data.resources_total_label()),
-                           VectorFromProtobuf(heartbeat_data.resources_total_capacity()));
-  ResourceSet remote_available(
-      VectorFromProtobuf(heartbeat_data.resources_available_label()),
-      VectorFromProtobuf(heartbeat_data.resources_available_capacity()));
-  ResourceSet remote_load(VectorFromProtobuf(heartbeat_data.resource_load_label()),
-                          VectorFromProtobuf(heartbeat_data.resource_load_capacity()));
-  // TODO(atumanov): assert that the load is a non-empty ResourceSet.
-  remote_resources.SetAvailableResources(std::move(remote_available));
-  // Extract the load information and save it locally.
-  remote_resources.SetLoadResources(std::move(remote_load));
+  if (!light_heartbeat_enabled_ || heartbeat_data.resources_total_label_size() > 0) {
+    ResourceSet remote_total(
+        VectorFromProtobuf(heartbeat_data.resources_total_label()),
+        VectorFromProtobuf(heartbeat_data.resources_total_capacity()));
+    remote_resources.SetTotalResources(std::move(remote_total));
+  }
+  if (!light_heartbeat_enabled_ || heartbeat_data.resources_available_label_size() > 0) {
```

Comment:
Can we pull the flag into an outer if statement to avoid checking it 3 times?

Reply:
Here are `or` condition but not `and`, so it's hard to pull out the flag.

Reply:
This could be changed like this:
```
if (light_heartbeat_enabled) {
  if (total_label_size > 0) {
    // set total resource
  }
  if (available_label_size > 0) {
   // set avail resource
  }
  if (load_label_size > 0) {
  // set load resource
  }
} else {
  // set total resource
  // set avail resource
  // set load resource
}
```
Will we add deprecated codes for making it clarity?